apps/modernization-api/src/main/java/gov/cdc/nbs/patient/PatientService.java

```diff
@@ -412,36 +417,162 @@ private void addRecordStatusQuery(Collection<RecordStatus> recordStatus, BoolQue
+            }
+            if (!isEmpty(input.getEducationLevelCode())) {
+                person.setEducationLevelCd(input.getEducationLevelCode());
+                modified = true;
+            }
+            if (!isEmpty(input.getPrimaryLanguageCode())) {
+                person.setPrimLangCd(input.getPrimaryLanguageCode());
+                modified = true;
+            }
+            if (!isEmpty(input.getSpeaksEnglishCode())) {
+                person.setSpeaksEnglishCd(input.getSpeaksEnglishCode());
+                modified = true;
+            }
+            if (!isEmpty(input.getEharsId())) {
+                person.setEharsId(input.getEharsId());
+                modified = true;
+            }
+            if (modified) {
+                person.setAsOfDateGeneral(input.getAsOf());
+            }
+            return personRepository.save(person);
+        });
+
         return sendPatientEvent(updateGeneralInfoEvent);
     }
 
     public PatientEventResponse addPatientName(NameInput input) {
         var user = SecurityUtil.getUserDetails();
         var event = NameInput.toAddRequest(user.getId(), getRequestId(), input);
+        personRepository.findById(input.getPatientId()).map(person -> {
+            PatientCommand.AddName addName = new PatientCommand.AddName(
+                    input.getPatientId(),
+                    input.getFirstName(),
+                    input.getMiddleName(),
+                    input.getLastName(),
+                    input.getSuffix(),
+                    PatientInput.NameUseCd.valueOf(input.getNameUseCd()),
+                    user.getId(),
+                    Instant.now()
+            );
+            person.add(addName);
+            return personRepository.save(person);
+        });
         return sendPatientEvent(event);
     }
 
     public PatientEventResponse updatePatientName(NameInput input) {
         var user = SecurityUtil.getUserDetails();
         var event = NameInput.toUpdateRequest(user.getId(), getRequestId(), input);
+        personRepository.findById(input.getPatientId()).map(person -> {
+            PatientCommand.AddName addName = new PatientCommand.AddName(
+                    input.getPatientId(),
+                    input.getFirstName(),
+                    input.getMiddleName(),
+                    input.getLastName(),
+                    input.getSuffix(),
+                    PatientInput.NameUseCd.valueOf(input.getNameUseCd()),
+                    user.getId(),
+                    Instant.now()
+            );
+            person.update(addName);
+            return personRepository.save(person);
+        });
         return sendPatientEvent(event);
     }
 
     public PatientEventResponse updateAdministrative(AdministrativeInput input) {
         var user = SecurityUtil.getUserDetails();
         var event = AdministrativeInput.toRequest(user.getId(), getRequestId(), input);
+        personRepository.findById(input.getPatientId()).map(person -> {
```

Comment:
sonarcloud tagging these maps

Reply:
`map` isn't terminal so they aren't doing anything, will have to use `ifPresent`